src/components/modals/export/MtsSysexExport.vue

```diff
@@ -0,0 +1,91 @@
+<script setup lang="ts">
+import MtsSysexExporter from "@/exporters/mts-sysex";
+import { sanitizeFilename } from "@/utils";
+import { ref } from "vue";
+import Modal from "@/components/ModalDialog.vue";
+import type { Scale } from "scale-workshop-core";
+import { clamp } from "xen-dev-utils";
+
+const props = defineProps<{
+  newline: string;
+  scaleName: string;
+  baseMidiNote: number;
+  scale: Scale;
+}>();
+
+const emit = defineEmits(["confirm", "cancel"]);
+
+function clampName(name: string): string {
+  return name.slice(0, 16);
+}
+
+function formatPresetIndex(input: string | number): number {
+  let number =
+    typeof input === "string" ? parseInt(input.replace(/\D/g, "")) : input;
+  if (Number.isNaN(number)) return 0;
+  return clamp(0, 127, number);
+}
+
+// Rarely implemented parameters
+// const deviceId = ref(0);
+// const bank = ref(0); (only for message 0x0804)
+
+const name = ref(clampName(props.scaleName));
```

Comment:
This update should hapen every time the modal is opened. Now this line of code basically reads like a statement of intent that masquerades as a computed property.

Reply:
I was able to add a watcher on `props` to solve this, but I'm not sure if this is the best way to implement input formatting. Do you think there's a better way? 
I tried an approach with a writable computed property (something I normally do in BootstrapVue) but couldn't get it to work, plus the Vue3 documentation says that should only be used in rare cases anyway.

Reply:
Yeah, it's a weird case because you want changes in the scale name to affect the clamped name but not the other way around. I'll check if there's a straightforward solution.

Reply:
This does what I'm after: Updating the name on the main interface changes what's in the modal, but modifying the modal text input doesn't reflect back on the main interface.
```Javascript
const name = ref("");

watch(
  () => props.scaleName,
  (newValue) => {
    name.value = clampName(newValue);
  },
  { immediate: true }
);
```

Reply:
Thanks so much! That's close to what I came up with too, good to have some validation on that.